### PR TITLE
fix alignment and missing script types for writing of `SerializedFile`

### PR DESCRIPTION
**Pull request by wd357dui**
for the `Write` method in `SerializedFile`:
 - add script types
 - `objectDataPosition` is aligned to 16 bytes
 - each object data is aligned to 8 bytes

Tested by reading and writing back (replacing) a game's `globalgamemanagers.assets` file (the game has unity version `2022.3.6f1`); with these script types and alignments changes, the game won't crash on startup anymore.

### CODE REVIEW

**Comment by wd357dui** [comment]
I have read the CLA Document and I hereby sign the CLA

File: Source/AssetRipper.IO.Files/SerializedFiles/SerializedFile.cs [hunk 1]
```diff
@@ -202,28 +205,20 @@ static ObjectInfo[] GetNewObjectInfoArray(ObjectInfo[]? objects)
...
-					case 2:
-						writer.Write((byte)0);
-						writer.Write((byte)0);
-						break;
-					case 3:
-						writer.Write((byte)0);
-						break;
+					writer.Write((byte)0);
 				}
 			}
```
Comment (ds5678) [inline Source/AssetRipper.IO.Files/SerializedFiles/SerializedFile.cs:223]: I'm thinking like:
```cs
static void AlignStream16(SerializedWriter writer)
{
	if ((writer.BaseStream.Position & 0b1111) != 0)
	{
		int padding = 16 - (int)(writer.BaseStream.Position & 0b1111);
		Span<byte> buffer = stackalloc byte[padding];
		buffer.Clear();
		writer.Write(buffer);
	}
}
```
or
```cs
[MethodImpl(MethodImplOptions.AggressiveInlining)]
static void AlignStream(SerializedWriter writer, [ConstantExpected] int alignment)
{
	Debug.Assert(alignment > 0);
	Debug.Assert(BitOperations.IsPow2(alignment));
	long bytesSinceLastAlignment = writer.BaseStream.Position & (alignment - 1);
	if (bytesSinceLastAlignment != 0)
	{
		int padding = alignment - (int)byteSinceLastAlignment;
		Span<byte> buffer = stackalloc byte[padding];
		buffer.Clear();
		writer.Write(buffer);
	}
}
```

**Review by ds5678 — CHANGES_REQUESTED** [review]
Thank you for your willingness to contribute!

**Comment by ds5678** [comment]
It sounds like this resolves #1276. Is that correct?

**Comment by wd357dui** [comment]
> It sounds like this resolves #1276. Is that correct?

Oh, I wasn't aware that there was an open issue for this, yes, after reading it, I can say that it's the exact issue down to the T!

> I also want details about how you know that 8 is required instead of 4.

I don't know about [DaXcess](https://github.com/DaXcess) but I figured it out by testing it with a game like I said (the same goes for script types and that 16-byte alignment).

**Review by ds5678 — APPROVED** [review]
Looks good to me. I'll merge as soon as the CI completes. Thank you for contributing!

**Pull request merged**